### PR TITLE
Updates to Existing Request page (Part 1)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#5767

*Description of changes:*

Commit 1:
- updated project version
- added base button SCSS (normal, primary, secondary)
- removed unneeded SCSS overrides (v-btn)
- added primary and secondary colours to SCSS theme
- updated layout etc on Existing Name Request page
- added common mixin
- updated Main Container layout/styles
- updated Refund Modal styles
- updated Vuetify base colours
- removed obsolete main container SASS
- removed obsolete submission tabs stop SASS
- added enums for Name State, NR Action and NR State
- updated in Existing Request Display component only
- changed request status to title case
- added a few NR states per designs
- updated base button styles
- simplified theme popular colors
- updated Existing Request Display page layout and content (WIP)
- adjusted Main Container layout
- reverted Vuetify theme changes
- moved some classes from main.sass to base.scss
- added some global styles
- added v-btn style overrides
- created Names Block component
- cleaned up a lot of styles on Existing Request Display page
- added some helpers to format text
- globally replaced fw-700 with font-weight-bold
- replaced some color hexes with names from theme
- misc cleanup

Commit 2:
- fixed main container (was broken at 960px breakpoint)
- updated whitespace on Existing Request Display page
- updated layout on Existing Request Search page so divs fit tightly (and therefore, outer padding is correct)
- updated layout on search page for consistency

Commit 3:
- moved names-block to existing request folder

Commit 4:
- added comment to mixin

Commit 5:
- deleted debugging code

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: client/src/assets/scss/base.scss
Comment (severinbeauvais) [inline client/src/assets/scss/base.scss:17]: I moved these to layout.scss.

File: client/src/assets/scss/base.scss
Comment (severinbeauvais) [inline client/src/assets/scss/base.scss:34]: I moved this to layout.scss.

File: client/src/assets/scss/overrides.scss
Comment (severinbeauvais) [inline client/src/assets/scss/overrides.scss:13]: These are just nested for cleanliness.

File: client/src/assets/scss/overrides.scss
Comment (severinbeauvais) [inline client/src/assets/scss/overrides.scss:35]: This is the global font size for buttons per Tracey (with some special cases).

Although this is a global style, there are misc overrides for this in various scss/sass/vue files. Please clean those up if you're working in those files.

File: client/src/assets/scss/theme.scss
Comment (severinbeauvais) [inline client/src/assets/scss/theme.scss:61]: There is some duplication here. Any volunteers to clean it up?

Also I see there are 2 greens. Tracey specified "app-green" for me. Did she also specify "approved" green? If so then we should ask her for a single value.

File: client/src/components/existing-request/existing-request-display.vue
Comment (severinbeauvais) [inline client/src/components/existing-request/existing-request-display.vue:207]: I'm still using this :)

I will remove in Part 2 commit for this feature.

File: client/src/components/existing-request/existing-request-display.vue
Comment (severinbeauvais) [inline client/src/components/existing-request/existing-request-display.vue:10]: Moved to new component.

File: client/src/components/existing-request/existing-request-display.vue
Comment (severinbeauvais) [inline client/src/components/existing-request/existing-request-display.vue:211]: Moved to new component.

File: client/src/components/new-request/main-container.vue
Comment (severinbeauvais) [inline client/src/components/new-request/main-container.vue:2]: Changes in this component were necessary for the proper layout of the Existing Request Display page.
However, this may have broken other pages that use Main Container. Can someone help me test and update those please?

File: client/src/enums/nr-state.ts
Comment (severinbeauvais) [inline client/src/enums/nr-state.ts:10]: I need to test these. Does anyone have a NR that's On Hold, and one that's Refunded? (I have asked John for these as well.)

File: client/src/plugins/vuetify.ts
Comment (severinbeauvais) [inline client/src/plugins/vuetify.ts:19]: These need to be kept in sync with our global colour variables.

File: client/src/sass/main.sass
Comment (severinbeauvais) [inline client/src/sass/main.sass:2]: I moved this to base.scss.

File: client/src/sass/main.sass
Comment (severinbeauvais) [inline client/src/sass/main.sass:80]: I replaced this with "font-weight-bold".

File: client/src/sass/main.sass
Comment (severinbeauvais) [inline client/src/sass/main.sass:171]: This button still exists but these styles are not needed.

File: client/src/sass/overrides.sass
Comment (severinbeauvais) [inline client/src/sass/overrides.sass:2]: This is a duplicate (was also in main.sass), so I removed it.

File: client/src/sass/theme.sass
Comment (severinbeauvais) [inline client/src/sass/theme.sass:1]: These variables are still used by components that use SASS.

Eventually these must die...

File: client/src/assets/scss/base.scss
Comment (severinbeauvais) [inline client/src/assets/scss/base.scss:13]: normal == 400 (or 500)

File: client/src/assets/scss/base.scss
Comment (severinbeauvais) [inline client/src/assets/scss/base.scss:27]: bold == 700

File: client/src/assets/scss/base.scss
Comment (severinbeauvais) [inline client/src/assets/scss/base.scss:71]: We have too many different gray background (and gray borders).

If you get a chance, please reduce this number.

File: client/src/assets/scss/theme.scss
Comment (severinbeauvais) [inline client/src/assets/scss/theme.scss:58]: Hmm, we have 2 greens for the same thing.
Tracey spec'ed out "app-green" using that hex code. Did she also spec out "approved" colour? We should ask her for a single approved green colour.

File: client/src/assets/scss/theme.scss
Comment (severinbeauvais) [inline client/src/assets/scss/theme.scss:59]: This is yet another background or border gray colour.

File: client/src/plugins/vuetify.ts
Comment (severinbeauvais) [inline client/src/plugins/vuetify.ts:18]: These Vuetify theme colours need to be kept in sync with our other theme colours. Please update as you go.

File: client/src/assets/scss/theme.scss
Comment (severinbeauvais) [reply client/src/assets/scss/theme.scss]: I checked with Tracey and she provided just 1 green :)

File: client/src/plugins/vuetify.ts
Comment (severinbeauvais) [reply client/src/plugins/vuetify.ts]: Obsolete.

File: client/src/assets/scss/theme.scss
Comment (severinbeauvais) [reply client/src/assets/scss/theme.scss]: Fixed.

**Review by vysakh-menon-aot — APPROVED** [review]

File: client/src/assets/scss/overrides.scss
Comment (cameron-freshworks) [reply client/src/assets/scss/overrides.scss:36]: Will definitely keep an eye on it after merge 👍

File: client/src/components/common/names-block.vue
Comment (cameron-freshworks) [inline client/src/components/common/names-block.vue:33]: Where do we use this? just for a little context

File: client/src/components/lower-info-area/lower-container.vue
Comment (cameron-freshworks) [inline client/src/components/lower-info-area/lower-container.vue:131]: It's too bad with urls you can't template string them, and do without the concatenation... or can you?!

File: client/src/components/mixins/common-mixin.ts
Comment (cameron-freshworks) [inline client/src/components/mixins/common-mixin.ts:5]: Maybe just a small comment would be cool :)

File: client/src/components/new-request/main-container.vue
Comment (cameron-freshworks) [reply client/src/components/new-request/main-container.vue:2]: oo yea, this will be far reaching. I can certainly help, but i've got a mountain of work in this current MRAS ticket 😅 

To be honest, i think the only major thing to consider is the max-width this sets throughout the app. Is that enforced some other way?

File: client/src/enums/name-state.ts
Comment (cameron-freshworks) [inline client/src/enums/name-state.ts:7]: Awesome!

File: client/src/enums/nr-action.ts
Comment (cameron-freshworks) [inline client/src/enums/nr-action.ts:11]: Double awesome

File: client/src/enums/nr-state.ts
Comment (cameron-freshworks) [reply client/src/enums/nr-state.ts:10]: I think* if you ask Sienna or Janis they can set any state you like, once you have an NR in draft.

**Review by cameron-freshworks — APPROVED** [review]
Really really good stuff, thanks for doing this!
One thing to check into is that max width, that container style removed held a lot of the sub components in size i think.
Good stuff!

File: client/src/assets/scss/overrides.scss
Comment (severinbeauvais) [reply client/src/assets/scss/overrides.scss:36]: Yeah, or you download my branch and have a look at the pages you're familiar with (please and thanks).

Comment (cameron-freshworks) [reply client/src/assets/scss/overrides.scss:36]: Can do!

File: client/src/components/common/names-block.vue
Comment (severinbeauvais) [reply client/src/components/common/names-block.vue:33]: This is the names block sub-component at the top of the Existing Request Display page. It looks like this:

image

Comment (severinbeauvais) [reply client/src/components/common/names-block.vue]: PS I just moved this sub-component from the common folder to the existing request folder.

File: client/src/components/lower-info-area/lower-container.vue
Comment (severinbeauvais) [reply client/src/components/lower-info-area/lower-container.vue:131]: Just tried. You can't (it adds spaces).

File: client/src/components/mixins/common-mixin.ts
Comment (severinbeauvais) [reply client/src/components/mixins/common-mixin.ts:6]: Done.

File: client/src/components/new-request/main-container.vue
Comment (severinbeauvais) [reply client/src/components/new-request/main-container.vue]: It looks OK to me atm, but I've only tested:
- New Name Search page
- Existing Name Request page
- Existing Request Display page

I'm assuming the padding will be off on the other pages, but responsive-wise it seems OK.

If anyone knows how to get to the other pages that use Main Container, please tell me!

File: client/src/enums/nr-state.ts
Comment (severinbeauvais) [reply client/src/enums/nr-state.ts:10]: Thanks for the tip (in case John falls through ;) ).

File: client/src/components/common/success.vue
Comment (severinbeauvais) [inline client/src/components/common/success.vue:35]: This is a global style, so redundant here.

File: client/src/components/existing-request/existing-request-search.vue
Comment (severinbeauvais) [inline client/src/components/existing-request/existing-request-search.vue:2]: OK, so, the bulk of the work in here is to untangle the div paddings/margins.
Now, you can click on an element (in dev tools) and you will see that divs do not have extra (or negative) margins. Everything is tight (except where whitespace is added for separation). This is the way it should be :)

File: client/src/components/new-request/main-container.vue
Comment (severinbeauvais) [inline client/src/components/new-request/main-container.vue:6]: There are now 2 spans inside this button: the first if edit mode, else the second one.

File: client/src/components/new-request/submit-request/timeout.vue
Comment (severinbeauvais) [inline client/src/components/new-request/submit-request/timeout.vue:33]: Again, redundant as there is already a global style for this class.

File: client/src/components/tabs.vue
Comment (severinbeauvais) [inline client/src/components/tabs.vue:26]: This caused whitespace at the wrong place.

**Pull request merged**
